PlutoHelperAgent/PreferencesWindowController.m

```diff
@@ -0,0 +1,210 @@
+//
+//  PreferencesWindowController.m
+//  PlutoHelperAgent
+//
+//  Created by Dave Allison on 26/02/2018.
+//  Copyright © 2018 Guardian News & Media. All rights reserved.
+//
+
+#import "PreferencesWindowController.h"
+
+@interface PreferencesWindowController ()
+
+@end
+
+@implementation PreferencesWindowController
+
+
+- (void) grab_data_from_keychain {
+    
+    UInt32 pwLength = 0;
+    
+    void* pwData = NULL;
+    
+    SecKeychainItemRef itemRef = NULL;
+    
+    NSString* service = @"PlutoHelperAgent";
+    
+    OSStatus pwAccessStatus = SecKeychainFindGenericPassword(
+                                                     
+                                                     NULL,         // Search default keychains
+                                                     
+                                                     (UInt32)service.length,
+                                                     
+                                                     [service UTF8String],
+                                                     
+                                                     0,
+                                                     
+                                                     NULL,
+                                                     
+                                                     &pwLength,
+                                                     
+                                                     &pwData,
+                                                     
+                                                     &itemRef      // Get a reference this time
+                                                     
+                                                     );
+    
+    if (pwAccessStatus == errSecSuccess) {
+        
+        NSData* data = [NSData dataWithBytes:pwData length:pwLength];
+        
+        NSString* password = [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];
+        
+        _PasswordText.stringValue = password;
```

Comment:
as below, for setting member varaibles with [this setPassword:password];

Reply:
That does not work so leaving current syntax.

Reply:
Apologies, I was reading quickly and assumed that _PasswordText is a string property of the controller which is then bound to the text control in Interface Builder.  Actually it's a reference to the text field, so either _PasswordText.stringValue or [self PasswordText].stringValue would be appropriate.  Does NSTextField have a specific setter method for stringValue?